src/org/zaproxy/zap/utils/TableExportButton.java

```diff
@@ -0,0 +1,134 @@
+import javax.swing.JTable;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+/**
+ * A {@code JButton} class to facilitate exporting tables to a file (such as CSV).
+ * 
+ * @since TODO add version
+ */
+public class TableExportButton extends JButton {
+
+	private static final long serialVersionUID = 3437613469695367668L;
+
+	private static final Logger LOGGER = Logger.getLogger(TableExportButton.class);
+	
+	private JTable exportTable = null;
+	
+	/**
+	 * Constructs a {@code TableExportButton} for the given table.
+	 * 
+	 * @param table the Table for which the data should be exported
+	 */
+	public TableExportButton(JTable table) {
+		super(Constant.messages.getString("export.button.name"));
+		setTable(table);
+		super.setIcon(new ImageIcon(TableExportButton.class.getResource("/resource/icon/16/115.png")));
+		super.addActionListener((new AbstractAction() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				WritableFileChooser chooser = new WritableFileChooser();
+				chooser.setSelectedFile(new File(Constant.messages.getString("export.button.default.filename")));
+				if (chooser.showSaveDialog(View.getSingleton().getMainFrame()) == WritableFileChooser.APPROVE_OPTION) {
+					String file = chooser.getSelectedFile().toString();
+					if (!file.endsWith(".csv")) {
+						file += ".csv";
+					}
+					try (CSVPrinter pw = new CSVPrinter(
+							Files.newBufferedWriter(chooser.getSelectedFile().toPath(), StandardCharsets.UTF_8),
+							CSVFormat.DEFAULT)) {
+						pw.printRecord(getColumnNames(getTable()));
+						int rowCount = getTable().getRowCount();
+						int colCount = getTable().getColumnCount();
+						for (int row = 0; row < rowCount; row++) {
+							List<Object> valueOfRow = new ArrayList<Object>();
+							for (int col = 0; col < colCount; col++) {
+								String value = getTable().getValueAt(row, col) == null ? "" : getTable().getValueAt(row, col).toString();
+								valueOfRow.add(value);
+							}
+							pw.printRecord(valueOfRow);
+						}
+						JOptionPane.showMessageDialog(View.getSingleton().getMainFrame(),
+								Constant.messages.getString("export.button.success"));
+					} catch (Exception ex) {
+						JOptionPane.showMessageDialog(View.getSingleton().getMainFrame(),
+								Constant.messages.getString("export.button.error") + "\n"
+										+ ex.getMessage());
+						LOGGER.error("Export Failed: " + ex.getMessage(), ex);
+					}
+				}
+			}
+		}));
+	}
+
+	/**
+	 * Gets a {@code List} of column names for the given table.
+	 * 
+	 * @param table the Table for which the column names should be returned
+	 * @return the {@code List} of column names
+	 */
+	private List<String> getColumnNames(JTable table) {

```

Comment:
The parameter could be removed and be used the instance variable (or the getter).